Tests/SourceKitLSPTests/SemanticTokensTests.swift

```diff
@@ -231,673 +98,916 @@ final class SemanticTokensTests: XCTestCase {
-        Token(line: 1, utf16index: 4, length: 3, kind: .function),
-        Token(line: 1, utf16index: 9, length: 1, kind: .number),
+    try await assertSemanticTokens(
+      markedContents: """
+        1️⃣func 2️⃣foo(3️⃣arg: 4️⃣Int) {}
+        5️⃣foo(6️⃣arg: 7️⃣1)
+        """,
+      expected: [
+        TokenSpec(marker: "1️⃣", length: 4, kind: .keyword),
+        TokenSpec(marker: "2️⃣", length: 3, kind: .identifier),
+        TokenSpec(marker: "3️⃣", length: 3, kind: .function),
+        TokenSpec(marker: "4️⃣", length: 3, kind: .struct, modifiers: .defaultLibrary),
+        TokenSpec(marker: "5️⃣", length: 3, kind: .function),
+        TokenSpec(marker: "6️⃣", length: 3, kind: .function),
+        TokenSpec(marker: "7️⃣", length: 1, kind: .number),
       ]
     )
   }
 
   func testFunctionDeclarationWithFirstAndSecondName() async throws {
-    let text = """
-      func foo(arg internalName: Int) {}
-      """
+    try await SkipUnless.sourcekitdHasSemanticTokensRequest()
 
-    let tokens = try await openAndPerformSemanticTokensRequest(text: text)
-    XCTAssertEqual(
-      tokens.tokens,
-      [
-        Token(line: 0, utf16index: 0, length: 4, kind: .keyword),
-        Token(line: 0, utf16index: 5, length: 3, kind: .identifier),
-        Token(line: 0, utf16index: 9, length: 3, kind: .function),
-        Token(line: 0, utf16index: 13, length: 12, kind: .identifier),
-        Token(line: 0, utf16index: 27, length: 3, kind: .struct, modifiers: .defaultLibrary),
+    try await assertSemanticTokens(
+      markedContents: """
+        1️⃣func 2️⃣foo(3️⃣arg 4️⃣internalName: 5️⃣Int) {}
+        """,
+      expected: [
+        TokenSpec(marker: "1️⃣", length: 4, kind: .keyword),
+        TokenSpec(marker: "2️⃣", length: 3, kind: .identifier),
+        TokenSpec(marker: "3️⃣", length: 3, kind: .function),
+        TokenSpec(marker: "4️⃣", length: 12, kind: .identifier),
+        TokenSpec(marker: "5️⃣", length: 3, kind: .struct, modifiers: .defaultLibrary),
       ]
     )
   }
 }
 
-extension Token {
-  fileprivate init(
-    line: Int,
-    utf16index: Int,
+fileprivate struct TokenSpec {
+  let marker: String
+  let length: Int
+  let kind: SemanticTokenTypes
+  let modifiers: SemanticTokenModifiers
+
+  init(
+    marker: String,
     length: Int,
     kind: SemanticTokenTypes,
     modifiers: SemanticTokenModifiers = []
   ) {
-    self.init(
-      start: Position(line: line, utf16index: utf16index),
-      utf16length: length,
-      kind: kind,
-      modifiers: modifiers
+    self.marker = marker
+    self.length = length
+    self.kind = kind
+    self.modifiers = modifiers
+  }
+}
+
+fileprivate func assertSemanticTokens(
+  markedContents: String,
+  language: Language = .swift,
+  range: (startMarker: String, endMarker: String)? = nil,
+  expected: [TokenSpec],
+  file: StaticString = #filePath,
+  line: UInt = #line
+) async throws {
+  try await SkipUnless.sourcekitdHasSemanticTokensRequest()
```

Comment:
Seems like each test case is already doing this, do we still need this?

Reply:
Oh, thanks for catching. I changed my mind about this half-way through and forgot to remove it from the assert function.